### PR TITLE
chore: rename `git_auth` to `external_auth` in our schema

### PR DESCRIPTION
**Pull request by kylecarbs**
We're changing Git auth to be external auth. It will support any OAuth2 or OIDC provider.

To split up the larger change I want to contribute the schema changes first, and I'll add the feature itself in another PR.

### CODE REVIEW

**Comment by kylecarbs** [comment]
@mafredri This should change zero logic in any code and no endpoints for the user. It just renames types and tables!

File: coderd/gitauth/config.go [hunk 1]
```diff
@@ -315,19 +319,19 @@ func ConvertConfig(entries []codersdk.GitAuthConfig, accessURL *url.URL) ([]*Con
...
-			ID:                  entry.ID,
-			Regex:               regex,
-			Type:                typ,
-			NoRefresh:           entry.NoRefresh,
-			ValidateURL:         entry.ValidateURL,
-			AppInstallationsURL: entry.AppInstallationsURL,
-			AppInstallURL:       entry.AppInstallURL,
+			OAuth2Config:           oauthConfig,
+			ID:                     entry.ID,
+			GitCloneRegex:          regex,
```
Comment (mafredri) [inline coderd/gitauth/config.go:329]: Should this be domain/auth url regex instead? Git and clone seems very specific.

File: coderd/gitauth/config.go [hunk 2]
```diff
@@ -315,19 +319,19 @@ func ConvertConfig(entries []codersdk.GitAuthConfig, accessURL *url.URL) ([]*Con
...
+			Type:                   typ,
+			NoRefresh:              entry.NoRefresh,
+			ValidateURL:            entry.ValidateURL,
+			GitAppInstallationsURL: entry.AppInstallationsURL,
+			GitAppInstallURL:       entry.AppInstallURL,
```
Comment (mafredri) [inline coderd/gitauth/config.go:334]: Why do we keep the git prefix here?

File: coderd/workspaceagents.go [hunk 3]
```diff
@@ -2191,7 +2191,7 @@ func (api *API) workspaceAgentsGitAuth(rw http.ResponseWriter, r *http.Request)
 		if len(api.GitAuthConfigs) > 0 {
```
Comment (mafredri) [inline coderd/workspaceagents.go:2191]: Rename this field?

**Review by mafredri — APPROVED** [review]
A few questions but otherwise looks good 👍

File: coderd/workspaceagents.go (hunk 3 above)
Comment (kylecarbs) [reply coderd/workspaceagents.go:2191]: Fixing!

File: coderd/gitauth/config.go (hunk 1 above)
Comment (kylecarbs) [reply coderd/gitauth/config.go:329]: Going to revert these renames for now. It'll make more sense in my future commit!

**Pull request merged**